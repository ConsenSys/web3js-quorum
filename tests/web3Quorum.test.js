const Web3 = require("web3");
const Web3Quorum = require("../src");
const { URL } = require("./tests-utils/constants");

describe("web3Quorum", () => {
  it("should match the web3Quorum priv namespace snapshot", () => {
    expect(Web3Quorum(new Web3(URL)).priv).toMatchSnapshot();
  });
  it("should match the web3Quorum eth.flexiblePrivacyGroup namespace snapshot", () => {
    expect(
      Web3Quorum(new Web3(URL)).eth.flexiblePrivacyGroup
    ).toMatchSnapshot();
  });
});
