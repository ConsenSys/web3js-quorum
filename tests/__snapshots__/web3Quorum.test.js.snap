// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web3Quorum should match the web3Quorum eth.flexiblePrivacyGroup namespace snapshot 1`] = `
Object {
  "findOnChainPrivacyGroup": [Function],
}
`;

exports[`web3Quorum should match the web3Quorum priv namespace snapshot 1`] = `
Object {
  "call": [Function],
  "createPrivacyGroup": [Function],
  "debugGetStateRoot": [Function],
  "deletePrivacyGroup": [Function],
  "distributeRawTransaction": [Function],
  "findPrivacyGroup": [Function],
  "generateAndDistributeRawTransaction": [Function],
  "generateAndSendRawTransaction": [Function],
  "generatePrivacyGroup": [Function],
  "getCode": [Function],
  "getEeaTransactionCount": [Function],
  "getFilterChanges": [Function],
  "getFilterLogs": [Function],
  "getLogs": [Function],
  "getPrivacyPrecompileAddress": [Function],
  "getPrivateTransaction": [Function],
  "getTransactionCount": [Function],
  "getTransactionReceipt": [Function],
  "newFilter": [Function],
  "sendRawTransaction": [Function],
  "subscribe": [Function],
  "subscribeWithPooling": [Function],
  "subscriptionPollingInterval": 1000,
  "uninstallFilter": [Function],
  "unsubscribe": [Function],
  "waitForTransactionReceipt": [Function],
}
`;
